hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,186 @@
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar r = pos.x * pos.x + pos.y * pos.y;
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((r < m_R02) - (r > m_R12));
+        // exit immediately if no collision is found or particle is not moving normal to the wall
+        // (since no new collision could have occurred if there is no normal motion)
+        if (sign == 0 || vel.x * vel.x + vel.y * vel.y == Scalar(0))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar R = (sign == 1) ? m_R02 : m_R12;
+        const Scalar a = vel.x * vel.x + vel.y * vel.y;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - R * R;
+        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * a * c))) / (Scalar(2) * a);
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to boundary conditions
+        // no-slip requires reflection of the tangential components.
+        if (sign == 1)
+            {
+            // no-slip requires reflection of the tangential components.
+            if (m_no_slip)
+                {
+                vel = -vel;
+                }
+            // slip requires reflection of the normal components.
+            else
+                {
+                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
+                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
+                }
+            }
+
+        if (sign == -1)
+            {
+            if (m_no_slip)
+                {
+                vel.x = -vel.x + Scalar(2) * m_V * -pos.y;
+                vel.y = -vel.y + Scalar(2) * m_V * pos.x;
+                vel.z = -vel.z;
+                }
+            else
+                {
+                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
+                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
+                }
+            }
+
+        return true;
+        }
+
+    //! Check if a particle is out of bounds
+    /*!
+     * \param pos Current particle position
+     * \returns True if particle is out of bounds, and false otherwise
+     */
+    HOSTDEVICE bool isOutside(const Scalar3& pos) const
+        {
+        const Scalar r = pos.x * pos.x + pos.y * pos.y;
+        return (r > m_R12 || r < m_R02);
+        }
+
+    //! Get inner radius of the cylinder
+    /*!
+     * \returns Inner radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR0() const
+        {
+        return m_R0;
```

Comment:
See comment below about why to do this:
```suggestion
        return slow::sqrt(m_R0_sq);
```
Can do the same for R1.